source/Halibut/Transport/IConnectionPool.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Diagnostics;
+
+namespace Halibut.Transport
+{
+    public interface IConnectionPool<TKey, TPooledResource> where TPooledResource : class, IPooledResource
+    {
+        int GetTotalConnectionCount();
+        TPooledResource Take(TKey endPoint);
+        Task<TPooledResource> TakeAsync(TKey endPoint, CancellationToken cancellationToken);
+        void Return(TKey endPoint, TPooledResource resource);
+        Task ReturnAsync(TKey endPoint, TPooledResource resource, CancellationToken cancellationToken);
+        void Clear(TKey key, ILog log = null);
```

Comment:
Can these sync methods be marked as `[Obsolete]`?

Reply:
Sure! Done!

Reply:
I decorated the actual methods as well